### PR TITLE
REPL: make alt-m on an empty prompt go to previous module

### PR DESCRIPTION
**Pull request by IanButterworth**
Continuation of https://github.com/JuliaLang/julia/pull/51616 https://github.com/JuliaLang/julia/pull/33872

Based on https://github.com/JuliaLang/julia/pull/51616#issuecomment-1783838395

cc. @tecosaur

### CODE REVIEW

**Comment by tecosaur** [comment]
Thanks for picking the up Ian!

File: stdlib/REPL/docs/src/index.md [hunk 1]
```diff
@@ -578,8 +578,10 @@ Main
 
 It is possible to change this contextual module via the function
 `REPL.activate(m)` where `m` is a `Module` or by typing the module in the REPL
-and pressing the keybinding Alt-m (the cursor must be on the module name). The `Main` module can be "activated" with an empty prompt plus the keybinding. The
-active module is shown in the prompt (unless it is `Main`):
+and pressing the keybinding Alt-m with the cursor on the module name (Esc-m on MacOS).
+Pressing the keybinding on an empty prompt returns to the previously activate
```
Comment (IanButterworth) [inline stdlib/REPL/docs/src/index.md:582]:
```suggestion
Pressing the keybinding on an empty prompt returns to the previously active
```

File: stdlib/REPL/src/LineEdit.jl [hunk 2]
```diff
@@ -1477,22 +1477,30 @@ end
 
 current_word_with_dots(s::MIState) = current_word_with_dots(buffer(s))
 
+previous_active_module = Main
```
Comment (KristofferC) [inline stdlib/REPL/src/LineEdit.jl:1480]: Maybe could be a typed global? (`::Module`)?

Comment (IanButterworth) [reply stdlib/REPL/src/LineEdit.jl:1480]: https://github.com/JuliaLang/julia/pull/52800

**Pull request merged**